### PR TITLE
discogs: Fix discogs_albumid extraction

### LINKED ISSUE #4225: discogs: discogs_albumid is always 0
Most of my music is imported using the discogs plugin (beets v1.6.0, formerly imported via beets 1.4.9), but the every single discogs_albumid field is equal to 0. I do notice, however, the mb_albumid field contains what should be in discogs_albumid.

E.g.
```
         data_source: Discogs
     discogs_albumid: 0
    discogs_artistid: 1954908
     discogs_labelid: 1085864
    mb_albumartistid: 1954908
          mb_albumid: 10346192
   mb_releasegroupid: 1185315
```

Reproduce this by importing an album and using Discogs as the source. Re-importing will do.

### PR DESCRIPTION
**Pull request by JOJ0**
## Description

- Fixes #4225
- Discogs changed their release URL format. This fixes the albumid extraction by using an improved extraction method.

## To Do

- [x] ~Documentation.~ 
- [x] Changelog.
- [x] Tests.

### CODE REVIEW

**Comment by sampsyo** [comment]
Looks good overall, except that a test seems to be failing.

Also, just to check: is `extract_release_id` now dead code (never called)? If so, we should remove it. ✨ 🐟

**Comment by JOJ0** [comment]
Hi @sampsyo
test.test_discogs.DGAlbumInfoTest.test_parse_minimal_release was failing because the test data does not contain a Discogs URL.

extract_release_id had a check whether an URI was passed at all and returned None immediately if not. I added such a check to extrac_release_id_regex as well.

**Review by sampsyo — COMMENTED** [review]
Ah, thanks for tracking down the failing test!

I made a silly suggestion within, but maybe the better fix would actually be to just fix the test so it actually includes an ID? (Unless we expect the API to ever return a response without a URL, which I doubt…)

File: beetsplug/discogs.py [hunk 1]
```diff
@@ -193,16 +193,17 @@ def extract_release_id_regex(album_id):
...
-                r'^\[?r?(?P<id>\d+)\]?$',
-                r'discogs\.com/release/(?P<id>\d+)-',
-                r'discogs\.com/[^/]+/release/(?P<id>\d+)',
-        ]:
-            match = re.search(pattern, album_id)
-            if match:
-                return int(match.group('id'))
-
-        return None
+        if album_id:
```
[the remark this answers is not in the document]
Comment (JOJ0) [reply beetsplug/discogs.py:196]: Well, yes, an early exit like that is much nicer to read! Thanks!

**Comment by JOJ0** [comment]
Thanks for the review! Added URI to test data, removed sanity check in `extract_release_id_regex()` and "rebased-in" master.

**Review by sampsyo — APPROVED** [review]
Yay; looks great!!

**Pull request merged**